src/Common/src/System/IO/RowConfigReader.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System.Diagnostics;
+
+namespace System.IO
+{
+    /// <summary> 
+    /// Helper for reading config files where each row is a key-value data pair.
+    /// The input key-values must not have any whitespace within them.
+    /// </summary>
+    internal struct RowConfigReader
```

Comment:
1. Why a `struct` and not a `class`?
2. This is a perfect candidate type for a Unit Test.